POSTFIX_CERTS_GENERATION.md

```diff
@@ -0,0 +1,102 @@
+# Steps for generation self signed SSL key for postfix
+Let's create a self signed SSL key in cert directory.
+In order to do it let's create `certs` directory.
+First generate a private key for the server
+
+```bash
+$ openssl genrsa -des3 -out localhost.key 2048
+```
+
+The output should be:
+```bash
+Generating RSA private key, 2048 bit long modulus
+..............................................................................................................+++
+.........................................................+++
+e is 65537 (0x010001)
+Enter pass phrase for localhost.key: <- ENTER PASSWORD
+Verifying - Enter pass phrase for localhost.key: ENTER PASSWORD
+
+```
+
+Now let's create a certificate request:
+```bash
+$ openssl req -new -key localhost.key -out localhost.csr
+```
+
+The output should be like:
+```bash
+$ openssl req -new -key localhost.key -out localhost.csr
+Enter pass phrase for localhost.key:
+You are about to be asked to enter information that will be incorporated
+into your certificate request.
+What you are about to enter is what is called a Distinguished Name or a DN.
+There are quite a few fields but you can leave some blank
+For some fields there will be a default value,
+If you enter '.', the field will be left blank.
+-----
+Country Name (2 letter code) [XX]:
+State or Province Name (full name) []:
+Locality Name (eg, city) [Default City]:
+Organization Name (eg, company) [Default Company Ltd]:
+Organizational Unit Name (eg, section) []:
+Common Name (eg, your name or your server's hostname) []: <- ENTER YOUR DOMAIN           
+Email Address []:
+
+Please enter the following 'extra' attributes
+to be sent with your certificate request
+A challenge password []: <- LEAVE EMPTY
+An optional company name []:
+
+```
+
+Now let's create a self signed key.
+```bash
+$ openssl x509 -req -days 365 -in localhost.csr -signkey localhost.key -out localhost.crt
+Signature ok
+subject=C = XX, L = Default City, O = Default Company Ltd, CN = localhost
+Getting Private key
+Enter pass phrase for localhost.key:
+
+```
+
+Let's remove password from private certificate.
+We do this, so we don't have to enter a password when you restart postfix.
+```bash
+$ openssl rsa -in localhost.key -out localhost.key.nopass
+$ mv localhost.key.nopass localhost.key
+```
+
+Make ourself a trusted CA. This is simplest option.
```

Comment:
What is the purpose of creating a private CA? You are not even using it to sign the certificate generated above.